driver.go

```diff
@@ -0,0 +1,59 @@
+package tcell
+
+import (
+	"errors"
+	"os"
+	"os/signal"
+	"syscall"
+)
+
+// ErrWinSizeUnused is for TermDrivers to signal to use the default platform
+// window size lookup method
+var ErrWinSizeUnused = errors.New("driver does not provide WinSize")
+
+// TermDriver allows you to customize the TTY used by Screen,
+// most notably to support a PTY pair that can be used with SSH servers.
+type TermDriver interface {
+	// Init sets up two file TTY/PTY file descriptors, which may be the same
+	// in some cases. It also takes a chan that is used to notify the Screen
+	// refresh the window size.
+	Init(winch chan os.Signal) (in *os.File, out *os.File, err error)
+
+	// Fini is called before the cleanup of Screen's Fini. It's typically used
+	// to unsubscribe window change signals.
+	Fini()
+
+	// WinSize returns the current window width and height. It can also return
+	// ErrWinSizeUnused to tell Screen to use platform syscalls to get the
+	// window size from the out file descriptor.
+	WinSize() (width int, height int, err error)
+}
+
+// defaultTermDriver is what's used when you don't specify a custom TermDriver
+type defaultTermDriver struct {
+	winch chan os.Signal
+	out   *os.File
+}
+
+func (d *defaultTermDriver) Init(winch chan os.Signal) (in *os.File, out *os.File, err error) {
+	in, err = os.OpenFile("/dev/tty", os.O_RDONLY, 0)
+	if err != nil {
+		return
+	}
+	out, err = os.OpenFile("/dev/tty", os.O_WRONLY, 0)
+	if err != nil {
+		return
+	}
+	signal.Notify(winch, syscall.Signal(0x1c))
```

Comment:
I would prefer this be symbolic, because 0x1c isn't going to be correct on all platforms.  This needs to be the signal SIGWINCH.  Or just not provided by default, but provided by linux, posix, etc. drivers.

Reply:
Perhaps the platform code should be responsible for setting up this signal handler?

Reply:
Thanks for the feedback! I wish it could be platform code, but we need to *not* do this in the case of SSH. You're totally right it shouldn't be hardcoded, I was assuming it would only matter if it were the platforms that it was correct for. Maybe we put this in a variable defined in platform code and reference that here?